src/main/java/org/broadinstitute/hellbender/tools/walkers/GnarlyGenotyper.java

```diff
@@ -0,0 +1,292 @@
+        headerLines.add(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.RAW_GENOTYPE_COUNT_KEY));
+        if (SUMMARIZE_PLs) {
+            headerLines.add(GATKVCFHeaderLines.getFormatLine(GATKVCFConstants.REFERENCE_GENOTYPE_QUALITY));
+            headerLines.add(GATKVCFHeaderLines.getFormatLine(GATKVCFConstants.GENOTYPE_QUALITY_BY_ALLELE_BALANCE));
+            headerLines.add(GATKVCFHeaderLines.getFormatLine(GATKVCFConstants.GENOTYPE_QUALITY_BY_ALT_CONFIDENCE));
+        }
+        final VCFHeader vcfHeader = new VCFHeader(headerLines, new TreeSet<>(sampleNameSet));
+        final VCFHeader dbHeader = new VCFHeader(headerLines);
+        vcfWriter.writeHeader(vcfHeader);
+        if (outputDbName != null) {
+            annotationDatabaseWriter.writeHeader(dbHeader);
+        }
+    }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    @Override
+    public void apply(VariantContext variant, ReadsContext reads, ReferenceContext ref, FeatureContext features) {
+        SimpleInterval variantStart = new SimpleInterval(variant.getContig(), variant.getStart(), variant.getStart());
+        //return early if there's no non-symbolic ALT since GDB already did the merging
+        if ( !variant.isVariant() || !GenotypeGVCFs.isProperlyPolymorphic(variant)
+                || variant.getAttributeAsInt(VCFConstants.DEPTH_KEY,0) == 0
+                || (onlyOutputCallsStartingInIntervals && !intervals.stream().anyMatch(interval -> interval.contains(variantStart)))) {
+            if (keepAllSites) {
+                VariantContextBuilder builder = new VariantContextBuilder(mqCalculator.finalizeRawMQ(variant));  //don't fill in QUAL here because there's no alt data
+                builder.filter(GATKVCFConstants.LOW_QUAL_FILTER_NAME);
+                builder.attribute(GATKVCFConstants.AC_ADJUSTED_KEY, 0);
+                vcfWriter.add(builder.make());
+            }
+            return;
+        }
+
+        //return early if variant doesn't meet QUAL threshold
+        if (!variant.hasAttribute(GATKVCFConstants.RAW_QUAL_APPROX_KEY)) {
+            warning.warn("Variant will not be output because it is missing the " + GATKVCFConstants.RAW_QUAL_APPROX_KEY + "key assigned by the ReblockGVCFs tool -- if the input did come from ReblockGVCFs, check the GenomicsDB vidmap.json annotation info");
+            return;
+        }
+
+        final VariantContextBuilder annotationDBBuilder = new VariantContextBuilder(variant);
+        final VariantContext finalizedVC = genotyperEngine.finalizeGenotype(variant, annotationDBBuilder);
+        if (finalizedVC != null && (!onlyOutputCallsStartingInIntervals || intervals.stream().anyMatch(interval -> interval.contains(variantStart)))) {
+            vcfWriter.add(finalizedVC);
+        }
+        if (annotationDatabaseWriter != null) {
+            annotationDatabaseWriter.add(annotationDBBuilder.make());  //we don't seem to have a sites-only option anymore, so do it manually
```

Comment:
Re: "we don't seem to have a sites-only option anymore"   There is actually a `--sites-only-vcf-output` argument in GATKTool..

Reply:
This code is so old that I think I wrote that before you added that option for me.